genetic.py

```diff
@@ -331,8 +455,16 @@ def _mutate(self, innovation_history_instance, existing_added_neurons):
         mutated_weights_innovation_history = self._mutate_connection_weights_random_walk(
             after_add_connection_innovation_history_instance)
 
+        self.log.add_weights_mutations_log()
+
+        # updated the new graph for this sample, but saved its identity (my_id)
+        sample.innovation_history_instance = mutated_weights_innovation_history
+        sample.graph = Graph(mutated_weights_innovation_history)
+
+        self.log.add_new_graph_creation_log()
+
         # return the new sample
-        return mutated_weights_innovation_history
+        # return mutated_weights_innovation_history
 
```

Comment:
Mutate comments:
Setting the graph manually seems odd, do you want to do this because you want to keep the sample id?
If so:
1. Instead of having 'is_final' in Sample, give an optional parameter "sample_id=None" that if specified this is the id to use.
2. Make mutate return the new sample. 
3. Revert changes in evolve to make the code take the returned Sample.

Reply:
1. I have 5 creations of sample one of them is called in random_walk func. 
for the all 4 other time call finalize after Sample creation?  
2. why should mutate return new sample? 
```suggestion
        for s in new_population:
            self._mutate(s, existing_added_neurons)
```
   already updates each sample in new_population
Or maybe you mean that crossover shouldnt create the finale Sample? i had to finalize the samples' id in crossover since we want to record the created sample in the log (and know how to track its parents)